### PR TITLE
Update NuGet config files to reduce Secure Supply Chain Analysis warnings

### PR DESCRIPTION
**Pull request by jonthysell**
This PR updates all used NuGet.Config files to reduce the number of warnings produced by the Secure Supply Chain Analysis task. This does not completely address them however.

Related issues: https://github.com/microsoft/react-native-windows/issues/9992, https://github.com/microsoft/react-native-windows/issues/9993, https://github.com/microsoft/react-native-windows/issues/9996, https://github.com/microsoft/react-native-windows/issues/9997

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9976)

### CODE REVIEW

**Comment by jonthysell** [comment]
Currently failing as the react-native-public feed does not upstream to NuGet.org. Investigating options.

**Comment by jonthysell** [comment]
Trying to resolve #9992 and #9993

**Comment by jonthysell** [comment]
It looks like we won't be able to make this work entirely, reducing the scope. See #9996, #9997.

**Comment by jonthysell** [comment]
I have re-based and reduced the number of changes as much as possible to avoid breaking functionality.

File: packages/playground/windows/NuGet.Config [hunk 1]
```diff
@@ -3,11 +3,4 @@
   <config>
     <add key="repositoryPath" value="packages" />
   </config>
-  <packageSources>
-    <clear />
-    <add key="Nuget.org" value="https://api.nuget.org/v3/index.json" />
-  </packageSources>
-  <disabledPackageSources>
```
Comment (dannyvv) [inline packages/playground/windows/NuGet.Config:10]: This is interesting as this was required to be injected? Is a new Nuget Version 'better' at not including random nuget.config files from all over the place and affecting determinisitc builds?

**Review by dannyvv — APPROVED** [review]

File: packages/playground/windows/NuGet.Config (hunk 1 above)
Comment (jonthysell) [reply packages/playground/windows/NuGet.Config:10]: Right, except nuget merges all the config files up the folder hierarchy. So by only specifying the sources in the root config, there's only one file for the task to complain about.

**Pull request merged**